pennylane/ops/op_math/adjoint_constructor.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule applies the symbolic operation that indicates the adjoint of an operator through the `adjoint` transform.
+"""
+from functools import wraps
+
+from pennylane.operation import Operator, AdjointUndefinedError
+from pennylane.queuing import QueuingContext, QueuingError
+from pennylane.tape import QuantumTape, stop_recording
+
+from .adjoint_class import Adjoint
+
+
+def _single_op_eager(op):
+    try:
+        return op.adjoint()
+    except AdjointUndefinedError:
+        return Adjoint(op)
+
+
+def _single_op_eager_update_queue(op):
+    try:
+        adj = op.adjoint()
+        try:
+            QueuingContext.update_info(op, owner=adj)
+        except QueuingError:
+            pass
+        QueuingContext.append(adj, owns=op)
+        return adj
+    except AdjointUndefinedError:
+        return Adjoint(op)
+
+
+# pylint: disable=no-member
+def adjoint(fn, lazy=True):
+    """Create the adjoint of an Operator or a function that applies the adjoint of the provided function.
+
+    Args:
+        fn (function or :class:`~.operation.Operator`): A single operator or a quantum function that
+            applies quantum operations.
+
+    Keyword Args:
+        lazy=True (bool): If the transform is behaving lazily, all operations are wrapped in a ``Adjoint`` class
+            and handled later. If ``lazy=False``, operation-specific adjoint decompositions are first attempted.
+
+    Returns:
+        (function or :class:`~.operation.Operator`): If an Operator is provided, returns an Operator that is the adjoint.
+        If a function is provided, returns a function with the same call signature that returns the Adjoint of the
+        provided function.
+
+    .. note::
+
+        While the adjoint and inverse are identical for Unitary gates, not all possible operators are Unitary.
+        This transform can also act on ``Channel``'s and ``Hamiltonian``'s, for which the inverse and adjoint are different.
+
+    .. seealso:: :class:`~.ops.arithmetic.Adjoint` and :meth:`~.operation.Operator.adjoint`
+
+    **Example**
+
+    This transform can also accept a single operator.
```

Comment:
```suggestion
    The adjoint transform can accept a single operator.
```

Sounded like a previous example is missing